src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -3483,6 +3483,16 @@ private void EmitConditionalOperator(BoundConditionalOperator expr, bool used)
         {
             Debug.Assert(expr.ConstantValueOpt == null, "Constant value should have been emitted directly");
 
+            // Generate branchless IL for (b ? 1 : 0).
+            if (used &&
+                hasIntegralValueZeroOrOne(expr.Consequence, out var isConsequenceOne) &&
+                hasIntegralValueZeroOrOne(expr.Alternative, out var isAlternativeOne) &&
+                isConsequenceOne != isAlternativeOne)
+            {
+                EmitCondExpr(expr.Condition, sense: isConsequenceOne);
```

Comment:
>EmitCondExpr(expr.Condition, sense: isConsequenceOne); [](http://example.com/codeflow?start=16&length=54)

Another thing to consider is the fact that the following is a valid IL (note, the method doesn't return 1 or  0):
```
.method public hidebysig static bool  Test() cil managed
{
  // Code size       2 (0x2)
  .maxstack  8
  IL_0000:  ldc.i4.m1
  IL_0001:  ret
} // end of method C1::Test
```

Are we going to optimize ```Test() ? 1 : 0```? 
If so, what result are we going to get and is it going to be different for a non-optimized code gen?

Reply:
Good catch, this was indeed broken. Added a test and fixed.